sig-autoscaling/charter.md

```diff
@@ -0,0 +1,74 @@
+# SIG Autoscaling Charter
+
+This charter adheres to the conventions described in the [Kubernetes Charter README] and uses
+the Roles and Organization Management outlined in [sig-governance].
+
+## Scope
+
+Covers development and maintenance of componets for automated scaling in
+Kubernetes.  This includes automated vertical and horizontal pod
+autoscaling, initial resource estimation, cluster-proportional system
+component autoscaling, and autoscaling of Kubernetes clusters themselves.
+
+### In scope
+
+- Autoscaling-related API objects, such as the HorizontalPodAutoscaler and
+  VerticalPodAutoscaler
+
+- Autoscaling-related tools, such as the cluster autoscaler and
+  cluster-proportional scaling tools (e.g. pod-nanny)
+
+- Ensuring API interfaces (the scale subresource) are availble and usable
+  to enable other SIG to write autoscalable objects, and enable people to
+  interact with those interfaces.
```

Comment:
I'd argue this goes requires a lot of apimachinery know-how, which is not too closely related to anything else we do in sig. I suspect you're the only one who actually knows how the polymorphic scale client actually works.

Reply:
Sure, but we're responsible for maintaining the HPA, and the HPA can't work w/o the scale subresource, and the scale subresource's object is part of the autoscaling API group.

@kubernetes/sig-api-machinery-misc any opinions?

Reply:
In general, api-machinery owns mechanisms, specific sigs own specific APIs.

In this case, I'd expect api-machinery to own the apiserver code that enables subresources, and autoscaling to own the in-tree scale subresource implementations, the autoscaling/v1 Scale API types, and custom code in the scale client.

Reply:
ack, that's more-or-less what I expected